lib/pdk/cli/util.rb

```diff
@@ -83,12 +83,16 @@ def module_version_check
       module_function :module_version_check
 
       def puppet_from_opts_or_env(opts)
+        use_puppet_dev = (opts || {})[:'puppet-dev'] || ENV['PDK_PUPPET_DEV']
         desired_puppet_version = (opts || {})[:'puppet-version'] || ENV['PDK_PUPPET_VERSION']
         desired_pe_version = (opts || {})[:'pe-version'] || ENV['PDK_PE_VERSION']
 
         begin
           puppet_env =
-            if desired_puppet_version
+            if use_puppet_dev
+              PDK::Util::PuppetVersion.fetch_puppet_dev
```

Comment:
I don't think it's optimal for `puppet_from_opts_or_env` to have the side effect of cloning/fetching the repo every time it's called. Could this be exposed separately and maybe this method can raise if the local repo doesn't exist?

Reply:
Yeah, it currently is a 1-1 correlation with the places you want to call it, but I think it's just unclear that calling that method will have that side effect, so I'd rather see explicit calls in those places.

Reply:
Ah, that's fair. I can make that change and at this part, I'll just call `.puppet_dev_env` to get the hash.